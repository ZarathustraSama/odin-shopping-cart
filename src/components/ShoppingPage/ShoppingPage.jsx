import { useEffect, useState } from "react";
import PropTypes from "prop-types";

import NavigationBar from "../NavigationBar/NavigationBar";
import ProductCard from "../ProductCard/ProductCard";

import './ShoppingPage.module.css';

function ShoppingPage({ shoppingCart = [] }) {
  const [products, setProducts] = useState([]);
  const [cart, setCart] = useState(shoppingCart);

  useEffect(() => {
    fetch('https://fakestoreapi.com/products?limit=10')
      .then(response => response.json())
      .then(json => setProducts(json))
      .catch(error => console.log(error))
    
  }, [])

  return (
    <>
      <NavigationBar shoppingCart={cart} checkout={true} />
      <main>
        {products.map(product => {
          return <ProductCard key={product.id} product={product} addToCartFn={setCart}></ProductCard>
        })}
      </main>
    </>
  )
}

ShoppingPage.propTypes = {
  shoppingCart: PropTypes.array.isRequired,
}

export default ShoppingPage;