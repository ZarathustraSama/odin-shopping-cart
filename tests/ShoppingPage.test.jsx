import { describe, it, expect } from 'vitest';
import { render } from "@testing-library/react";
import { BrowserRouter } from 'react-router-dom';

import ShoppingPage from '../src/components/ShoppingPage/ShoppingPage';

describe("ShoppingPage", () => {
  it("should render a NavBar component, and a list of items fetched, each put in a card", () => {
    const { container } = render(<BrowserRouter><ShoppingPage /></BrowserRouter>);
    expect(container).toMatchSnapshot();
  })
})