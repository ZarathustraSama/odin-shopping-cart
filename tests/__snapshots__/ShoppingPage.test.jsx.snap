// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ShoppingPage > should render a NavBar component before fetching 1`] = `
<div>
  <nav>
    <a
      data-discover="true"
      href="/"
    >
      Home
    </a>
    <a
      data-discover="true"
      href="/shop"
    >
      Shop
    </a>
    <span
      role="textbox"
    >
      Your cart currently has 
      0
       items. 
    </span>
    <button
      class="checkoutBtn"
    >
      Checkout
    </button>
  </nav>
  <main />
</div>
`;
